tests/arch/riscv/pmp/no-multithreading/testcase.yaml

```diff
@@ -0,0 +1,13 @@
+common:
+  platform_allow:
+    - qemu_riscv32
+    - qemu_riscv32e
+    - qemu_riscv64
+  filter: CONFIG_RISCV_PMP
+  ignore_faults: true
+
+tests:
+  arch.riscv.pmp.no-mt.isr-stack-guard:
+    extra_args: EXTRA_CPPFLAGS=-DVIA_TWISTER=0
```

Comment:
... but if we want to go this route, then I'd rather use `EXTRA_CFLAGS` here and change `VIA_TWISTER` to something sounding less generic